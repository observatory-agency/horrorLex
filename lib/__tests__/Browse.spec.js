const Browse = require('../Browse');

jest.mock('mongodb');

describe('Browse', () => {
  let browse;
  const model = {
    collectionName: 'films',
    aggregate: jest.fn(() => ({
      toArray: jest.fn(() => [{
        _id: { title: 'book1' },
      }]),
    })),
  };

  beforeEach(() => {
    browse = new Browse(model);
  });

  describe('constructor', () => {
    it('should set this.model', () => {
      expect(browse.model).toBe(model);
    });
  });

  describe('createRegexStr', () => {
    describe('"a-zA-Z" character', () => {
      it('should return a formatted regex expressions', () => {
        const validChar = 'a';
        expect(Browse.createRegexStr(validChar)).toMatchSnapshot();
      });
    });
    describe('numerical character', () => {
      it('should return a formatted regex expressions', () => {
        const validChar = 'num';
        expect(Browse.createRegexStr(validChar)).toMatchSnapshot();
      });
    });
    describe('invalid characters', () => {
      it('should throw', () => {
        const invalidChar = '<';
        expect(() => {
          Browse.createRegexStr(invalidChar);
        }).toThrow('Invalid characters');
      });
    });
  });

  describe('byChar', () => {
    it('should return an object of formatted results', async () => {
      const regexChar = Browse.createRegexStr('a');
      const results = await browse.byChar(regexChar);
      expect(results).toMatchSnapshot();
    });
    describe('when there are tags included with the results', () => {
      beforeEach(() => {
        browse = new Browse({
          collectionName: 'films',
          aggregate: jest.fn(() => ({
            toArray: jest.fn(() => [{
              _id: { tags: ['tag1', 'tag2'], title: 'book1' },
            }]),
          })),
        });
      });
      it('should return an object of formatted results', async () => {
        const regexChar = Browse.createRegexStr('a');
        const results = await browse.byChar(regexChar);
        expect(results).toMatchSnapshot();
      });
    });
  });
});
