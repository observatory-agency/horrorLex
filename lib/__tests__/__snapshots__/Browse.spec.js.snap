// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browse byChar should return an object of formatted results 1`] = `
Array [
  Object {
    "title": "book1",
    "uriEncoded": Object {
      "tags": null,
      "title": "book1",
    },
  },
]
`;

exports[`Browse byChar when there are tags included with the results should return an object of formatted results 1`] = `
Array [
  Object {
    "title": "book1",
    "uriEncoded": Object {
      "tags": "tag1%2Ctag2",
      "title": "book1",
    },
  },
]
`;

exports[`Browse createRegexStr "a-zA-Z" character should return a formatted regex expressions 1`] = `"^[Aa]"`;

exports[`Browse createRegexStr numerical character should return a formatted regex expressions 1`] = `"^[0-9]"`;
